processors/base/helpers.py

```diff
@@ -202,9 +202,12 @@ def find_trial_by_identifiers(conn, identifiers):
     for source, identifier in identifiers.items():
         query = QUERY % json.dumps({source: identifier})
         records = list(conn['database'].query(query))
-        if not records:
-            continue
-        trial = conn['database']['trials'].find_one(id=records[0]['trial_id'].hex)
+        for record in records:
+            if not record['trial_id']:
+                continue
+            trial = conn['database']['trials'].find_one(id=record['trial_id'].hex)
+            if trial:
+                break
```

Comment:
Thanks. I'll try it. 

But now I'm trying to focus only on business logic (like more queries but easier to understand what code does).